diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-40dd6610341962b6 b/.bun-tag-40dd6610341962b6
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-6a87d334ce6b81ae b/.bun-tag-6a87d334ce6b81ae
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-910ee77fcb7bbb05 b/.bun-tag-910ee77fcb7bbb05
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-92850e0b33f189ed b/.bun-tag-92850e0b33f189ed
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-98595f30a74d64ae b/.bun-tag-98595f30a74d64ae
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-a11d7b66f87737aa b/.bun-tag-a11d7b66f87737aa
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-be2dbda66f0762de b/.bun-tag-be2dbda66f0762de
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-c22bf20fe8a62757 b/.bun-tag-c22bf20fe8a62757
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-c7bb2297719194bf b/.bun-tag-c7bb2297719194bf
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-cab7e1b76c1415b4 b/.bun-tag-cab7e1b76c1415b4
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-f862effb3d244c60 b/.bun-tag-f862effb3d244c60
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/dist/server/routeHandler/index.js b/dist/server/routeHandler/index.js
index 59e5d2087f55396a43699cc342736d74c67b806d..1924c1d56e4782ad3e839f0ddbfa58fbac28bfde 100644
--- a/dist/server/routeHandler/index.js
+++ b/dist/server/routeHandler/index.js
@@ -3,6 +3,7 @@ import * as server from 'next/server.js';
 import { tokenHandler } from './tokenHandler.js';
 export async function GetFn(req) {
     const { pathname } = new URL(req.url);
+    console.log("Server:", server)
     switch (pathname) {
         case '/api/faust/token/':
         case '/api/faust/token': {
diff --git a/dist/server/routeHandler/tokenHandler.js b/dist/server/routeHandler/tokenHandler.js
index 236a4912cc0e775155ead38a8b2349ee48c0f3ad..3586c44cb3a8f3347ceda8b87310dc313d72621d 100644
--- a/dist/server/routeHandler/tokenHandler.js
+++ b/dist/server/routeHandler/tokenHandler.js
@@ -1,5 +1,6 @@
 import { cookies } from 'next/headers.js';
 import { getWpUrl, getWpSecret } from '../../faust-core-utils.js';
+
 export async function tokenHandler(req, s) {
     var _a, _b;
     try {
@@ -59,9 +60,18 @@ export async function tokenHandler(req, s) {
          * @TODO Set the refresh token cookie with the new refresh token
          * and expiration.
          */
+        console.log("server", s)
+        console.log("data", data)
+
         const res = new s.NextResponse(JSON.stringify(data), {
             status: 200,
+            headers: {
+                'Content-Type': 'application/json',
+            },
         });
+
+        console.log("res", res)
+        
         res.cookies.set(cookieName, data.refreshToken, {
             secure: true,
             httpOnly: true,
