diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-40dd6610341962b6 b/.bun-tag-40dd6610341962b6
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-5274a5012462e053 b/.bun-tag-5274a5012462e053
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-54ff5039d7429f0f b/.bun-tag-54ff5039d7429f0f
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-55fa145ee4e98ebb b/.bun-tag-55fa145ee4e98ebb
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-6a87d334ce6b81ae b/.bun-tag-6a87d334ce6b81ae
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-83a6a084bc4e9acb b/.bun-tag-83a6a084bc4e9acb
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-910ee77fcb7bbb05 b/.bun-tag-910ee77fcb7bbb05
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-92850e0b33f189ed b/.bun-tag-92850e0b33f189ed
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-98595f30a74d64ae b/.bun-tag-98595f30a74d64ae
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-a11d7b66f87737aa b/.bun-tag-a11d7b66f87737aa
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-bdd28e56d5bc6e6f b/.bun-tag-bdd28e56d5bc6e6f
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-be2dbda66f0762de b/.bun-tag-be2dbda66f0762de
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-c22bf20fe8a62757 b/.bun-tag-c22bf20fe8a62757
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-c7bb2297719194bf b/.bun-tag-c7bb2297719194bf
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-cab7e1b76c1415b4 b/.bun-tag-cab7e1b76c1415b4
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-cb9cb366b7fc04bf b/.bun-tag-cb9cb366b7fc04bf
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/@faustwp/experimental-app-router/.bun-tag-f862effb3d244c60 b/.bun-tag-f862effb3d244c60
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/dist/server/routeHandler/tokenHandler.js b/dist/server/routeHandler/tokenHandler.js
index 236a4912cc0e775155ead38a8b2349ee48c0f3ad..9b7b275dcb63736cb4f6e520a36ec99ee154f17a 100644
--- a/dist/server/routeHandler/tokenHandler.js
+++ b/dist/server/routeHandler/tokenHandler.js
@@ -1,5 +1,7 @@
 import { cookies } from 'next/headers.js';
 import { getWpUrl, getWpSecret } from '../../faust-core-utils.js';
+import { NextResponse } from 'next/server';
+
 export async function tokenHandler(req, s) {
     var _a, _b;
     try {
@@ -32,21 +34,16 @@ export async function tokenHandler(req, s) {
                 refreshToken,
             }),
         });
+
+        // Log response status and body
+        console.log('Response status:', response.status);
+        const responseBody = await response.text();
+        console.log('Response body:', responseBody);
+
         if (!response.ok) {
-            /**
-             * Remove the refresh token from the cookie in the case the token is:
-             * - expired
-             * - invalid
-             * - revoked
-             * - from a different WordPress instance when developing on localhost
-             */
-            /**
-             * @TODO Delete the cookie
-             */
+            // @TODO Delete the cookie
             // cookieStore.delete(cookieName);
-            /**
-             * @TODO throw different errors based on response
-             */
+            // @TODO throw different errors based on response
             return new Response(JSON.stringify({ error: 'Unauthorized' }), {
                 status: 401,
                 headers: {
@@ -54,14 +51,16 @@ export async function tokenHandler(req, s) {
                 },
             });
         }
-        const data = (await response.json());
-        /**
-         * @TODO Set the refresh token cookie with the new refresh token
-         * and expiration.
-         */
-        const res = new s.NextResponse(JSON.stringify(data), {
+
+        const data = JSON.parse(responseBody);
+
+        const res = new NextResponse(JSON.stringify(data), { // Ensure correct usage
             status: 200,
         });
+
+        console.log("data", data);
+        console.log("res", res);
+
         res.cookies.set(cookieName, data.refreshToken, {
             secure: true,
             httpOnly: true,
@@ -69,9 +68,9 @@ export async function tokenHandler(req, s) {
             expires: new Date(data.refreshTokenExpiration * 1000),
             sameSite: 'lax',
         });
+
         return res;
-    }
-    catch (err) {
+    } catch (err) {
         console.error('Invalid response for authorize handler:', err);
         return new Response(JSON.stringify({ error: 'Internal Server Error' }), {
             status: 500,
